services/wallet/datastore.go

```diff
@@ -616,17 +614,29 @@ func (pg *Postgres) LinkWallet(ctx context.Context, id string, userDepositDestin
 	return nil
 }
 
-// CustodianLink - representation of wallet_custodian record
+// TODO(clD11): CustodianLink represent a wallet_custodian. Review during wallet refactor
+
+// CustodianLink representation a wallet_custodian record.
 type CustodianLink struct {
-	WalletID           *uuid.UUID `json:"wallet_id" db:"wallet_id" valid:"uuidv4"`
-	Custodian          string     `json:"custodian" db:"custodian" valid:"in(uphold,brave,gemini,bitflyer)"`
-	CreatedAt          time.Time  `json:"created_at" db:"created_at" valid:"-"`
-	UpdatedAt          *time.Time `json:"updated_at" db:"updated_at" valid:"-"`
-	LinkedAt           time.Time  `json:"linked_at" db:"linked_at" valid:"-"`
-	DisconnectedAt     *time.Time `json:"disconnected_at" db:"disconnected_at" valid:"-"`
-	DepositDestination string     `json:"deposit_destination" db:"deposit_destination" valid:"-"`
-	LinkingID          *uuid.UUID `json:"linking_id" db:"linking_id" valid:"uuid"`
-	UnlinkedAt         *time.Time `json:"unlinked_at" db:"unlinked_at" valid:"-"`
+	WalletID       *uuid.UUID `json:"wallet_id" db:"wallet_id" valid:"uuidv4"`
+	Custodian      string     `json:"custodian" db:"custodian" valid:"in(uphold,brave,gemini,bitflyer)"`
+	CreatedAt      time.Time  `json:"created_at" db:"created_at" valid:"-"`
+	UpdatedAt      *time.Time `json:"updated_at" db:"updated_at" valid:"-"`
+	LinkedAt       time.Time  `json:"linked_at" db:"linked_at" valid:"-"`
+	DisconnectedAt *time.Time `json:"disconnected_at" db:"disconnected_at" valid:"-"`
+	LinkingID      *uuid.UUID `json:"linking_id" db:"linking_id" valid:"uuid"`
+	UnlinkedAt     *time.Time `json:"unlinked_at" db:"unlinked_at" valid:"-"`
+}
+
+// TODO(clD11): Wallet Refactor. These should not be nullable, fix pointers and raname fields for consistency.
+
+func NewSolanaCustodialLink(walletID uuid.UUID, depositDestination string) *CustodianLink {
+	const depositProviderSolana = "solana"
+	return &CustodianLink{
+		WalletID:  &walletID,
+		LinkingID: ptrFromUUID(uuid.NewV5(ClaimNamespace, depositDestination)),
```

Comment:
Nit: There's our recently added `ptr.To` thing.

Reply:
I thought the we were deprecating the use of the ptr package as per a previous PR comment. Until we resolve the issues with generics and the make commands we are going analogue :smile:

Reply:
Oh yes, OK then. 😄